resources/config/varnish/fos_purge.vcl

```diff
@@ -12,6 +12,17 @@ sub fos_purge_recv {
         if (!client.ip ~ invalidators) {
             return (synth(405, "Not allowed"));
         }
+
+        # If http header "xkey" is set, we assume purge is on key and you have Varnish xkey installed
+        if (req.http.xkey) {
```

Comment:
Definitely make this configurable in on the PHP side. That’s what we did for the tags header, too.